src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATask.cs

```diff
@@ -52,7 +53,9 @@ public override void Initialize(AnalysisContext context)
                     return;
                 }
 
-                var configuredAsyncDisposable = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeCompilerServicesConfiguredAsyncDisposable);
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
```

Comment:
It might be a good idea to move this variable before `TryGetTaskTypes`, and pass it instead of passing context.Compilation.